plugins/convertStyleToAttrs.js

```diff
@@ -63,70 +65,77 @@ var stylingProps = require('./_collections').attrsGroups.presentation,
  *             ⬇
  * <g fill="#000" color="#fff" style="-webkit-blah: blah">
  *
- * @param {Object} item current iteration item
- * @return {Boolean} if false, item will be filtered out
- *
  * @author Kir Belevich
+ *
+ * @type {import('../lib/types').Plugin<{
+ *   keepImportant?: boolean
+ * }>}
  */
-exports.fn = function (item, params) {
-  if (item.type === 'element' && item.attributes.style != null) {
-    // ['opacity: 1', 'color: #000']
-    let styles = [];
-    const newAttributes = {};
+exports.fn = (_root, params) => {
+  const { keepImportant = false } = params;
+  return {
+    element: {
+      enter: (node) => {
+        if (node.attributes.style != null) {
+          // ['opacity: 1', 'color: #000']
+          let styles = [];
+          /**
+           * @type {Record<string, string>}
+           */
+          const newAttributes = {};
+
+          // Strip CSS comments preserving escape sequences and strings.
+          const styleValue = node.attributes.style.replace(
+            regStripComments,
+            (match) => {
+              return match[0] == '/'
+                ? ''
+                : match[0] == '\\' && /[-g-z]/i.test(match[1])
+                ? match[1]
+                : match;
+            }
+          );
+
+          regDeclarationBlock.lastIndex = 0;
+          // eslint-disable-next-line no-cond-assign
+          for (var rule; (rule = regDeclarationBlock.exec(styleValue)); ) {
+            if (!keepImportant || !rule[3]) {
+              styles.push([rule[1], rule[2]]);
+            }
+          }
 
-    // Strip CSS comments preserving escape sequences and strings.
-    const styleValue = item.attributes.style.replace(
-      regStripComments,
-      (match) => {
-        return match[0] == '/'
-          ? ''
-          : match[0] == '\\' && /[-g-z]/i.test(match[1])
-          ? match[1]
-          : match;
-      }
-    );
+          if (styles.length) {
+            styles = styles.filter(function (style) {
+              if (style[0]) {
+                var prop = style[0].toLowerCase(),
```

Comment:
Just curiosity here. Why the need to use var?

Reply:
I just kept old code here

> Note: review with hidden whitespaces